tester/pkg/config.go

```diff
@@ -0,0 +1,45 @@
+package pkg
+
+type TestConfig struct {
+	Cluster         *Cluster `yaml:"cluster"`
+	Region          string   `yaml:"region"`
+	BuildScript     string   `yaml:"build"`
+	InstallScript   string   `yaml:"install"`
+	UninstallScript string   `yaml:"uninstall"`
+	TestScript      string   `yaml:"test"`
+}
+
+type Cluster struct {
+	Kops *KopsCluster `yaml:"kops"`
+	Eks  *EksCluster  `yaml:"eks"`
+}
+
+type KopsCluster struct {
+	Region            string `yaml:"region"`
+	Zones             string `yaml:"zones"`
+	NodeCount         int    `yaml:"nodeCount"`
+	NodeSize          string `yaml:"nodeSize"`
+	KubernetesVersion string `yaml:"kubernetesVersion"`
+	FeatureGates      string `yaml:"featureGates"`
+	IamPolicies       string `yaml:"iamPolicies"`
+	//KubeApiServer     KubeApiServer `yaml:"kubeApiServer"`
+	//Kubelet           Kubelet       `yaml:"kubelet"`
+}
+
+type KubeApiServer struct {
+	FeatureGates FeatureGates `yaml:"featureGates"`
+}
+
+type FeatureGates struct {
```

Comment:
prefer to keep it map[string]string (same as kops config https://github.com/kubernetes/kops/blob/b0860f7738f8f1c38e288910494ef6382e4dd7fc/pkg/apis/kops/v1alpha1/componentconfig.go#L410) otherwise we'll be chasing every feature

In the future we could use the kops API directly, rather than duplicating its fields here...something like:
```
// kops.ClusterSpec is https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/cluster.go#L47
type KopsCluster struct {
    kops.ClusterSpec spec `yaml:"clusterSpec"` 
}
```
Then run `kops create cluster --config=FILE_FOR_KopsCluster.spec`

As opposed to creating kops cluster, getting the spec file, then appending to it.

But it's probably not worth the trouble to figure it out now.

Reply:
Good point on kops.ClusterSpec. We should try it later.
I created a FeatureGates string, which we will append to the cluster spec for now.